src/core/algorithms/fd/tane/tane.h

```diff
@@ -1,47 +1,69 @@
 #pragma once
 
-#include <string>
-
 #include "algorithms/fd/pli_based_fd_algorithm.h"
+#include "algorithms/fd/tane/lattice_level.h"
 #include "config/error/type.h"
+#include "enums.h"
+#include "model/table/column_data.h"
+#include "model/table/column_layout_relation_data.h"
 #include "model/table/position_list_index.h"
-#include "model/table/relation_data.h"
 
 namespace algos {
+namespace tane {
 
-class Tane : public PliBasedFDAlgorithm {
-private:
-    void RegisterOptions();
-    void MakeExecuteOptsAvailableFDInternal() final;
+class TaneCommon : public PliBasedFDAlgorithm {
+protected:
+    config::ErrorType max_fd_error_;
+    config::ErrorType max_ucc_error_;
 
+private:
     void ResetStateFd() final;
```

Comment:
let's write definition right here
```suggestion
    void ResetStateFd() final {}
```